dotcms-integration/src/test/java/com/dotmarketing/portlets/contentlet/business/ContentletAPITest.java

```diff
@@ -8525,4 +8527,251 @@ public void getMostRecentPublishedContent() throws Exception {
+                .nextPersisted();
+
+        ContentTypeDataGen.addField(field);
+
+        // publish contentlet
+        final ContentletDataGen contentletDataGen = new ContentletDataGen(contentType);
+        final String value = "LOoooooooooooooooooooooooong teeeeeeeeeeeeext";
+        contentletDataGen.setProperty(keyPropName, value);
+        final Contentlet publishedContentlet = contentletDataGen.nextPersisted();
+        ContentletDataGen.publish(publishedContentlet);
+        final String contentInode = publishedContentlet.getInode();
+
+        final Contentlet contentRetrieved = APILocator.getContentletAPI().find(contentInode, user, false);
+
+        assertNotNull(contentRetrieved);
+        assertEquals(contentInode, contentRetrieved.getInode());
+
+        final String valueRetrieved = (String) contentRetrieved.get(keyPropName);
+        assertEquals(value, valueRetrieved);
+    }
+
+    /**
+     * Method to test: {@link ESContentletAPIImpl#setContentletProperty(Contentlet, Field, Object)}
+     * When: Create a content type with date and title and sets to it
+     * Should: should be properly set, taking a Date as an input
+     */
+    @Test
+    public void test_setContentletProperty_date() throws Exception {
+
+        final ContentType contentType = new ContentTypeDataGen().nextPersisted();
+        final String keyPropName = "date";
+        ContentTypeDataGen.addField(new FieldDataGen()
+                .velocityVarName("title")
+                .contentTypeId(contentType.id())
+                .type(TextField.class)
+                .nextPersisted());
+
+        ContentTypeDataGen.addField(new FieldDataGen()
+                .velocityVarName(keyPropName)
+                .contentTypeId(contentType.id())
+                .type(DateField.class)
+                .defaultValue("now")
+                .nextPersisted());
+
+        // publish contentlet
+        final ContentletDataGen contentletDataGen = new ContentletDataGen(contentType);
+        final Date value = new Date();
+        contentletDataGen.setProperty(keyPropName, value);
+        final Contentlet publishedContentlet = contentletDataGen.nextPersisted();
+        ContentletDataGen.publish(publishedContentlet);
+        final String contentInode = publishedContentlet.getInode();
+
+        final Contentlet contentRetrieved = APILocator.getContentletAPI().find(contentInode, user, false);
+
+        assertNotNull(contentRetrieved);
+        assertEquals(contentInode, contentRetrieved.getInode());
+
+        final Date valueRetrieved = (Date) contentRetrieved.get(keyPropName);
+        assertEquals(value, valueRetrieved);
+    }
+
+    /**
+     * Method to test: {@link ESContentletAPIImpl#setContentletProperty(Contentlet, Field, Object)}
+     * When: Create a content type with float and title and sets to it
+     * Should: should be properly set, taking a float as an input
+     */
+    @Test
+    public void test_setContentletProperty_float() throws Exception {
```

Comment:
@jdotcms can we make sure we have test coverage for all the different strategies/field-types/field-data-tyes? system, boolean, etc. We should have every single strategy covered with tests.